test/functional/http/post/3.votes.js

```diff
@@ -0,0 +1,437 @@
+'use strict';
+
+var node = require('../../../node');
+var shared = require('../../shared');
+var constants = require('../../../../helpers/constants');
+
+var sendTransactionPromise = require('../../../common/apiHelpers').sendTransactionPromise;
+var creditAccountPromise = require('../../../common/apiHelpers').creditAccountPromise;
+var registerDelegatePromise = require('../../../common/apiHelpers').registerDelegatePromise;
+var waitForBlocksPromise = node.Promise.promisify(node.waitForBlocks);
+
+describe('POST /api/transactions (type 3) votes', function () {
+
+	var badTransactions = [];
+	var goodTransactions = [];
+	var badTransactionsEnforcement = [];
+	var goodTransactionsEnforcement = [];
+
+	var account = node.randomAccount();
+	var accountNoFunds = node.randomAccount();
+	var accountMinimalFunds = node.randomAccount();
+	var accountDuplicate = node.randomAccount();
+
+	var account33 = node.randomAccount();
+	var delegates33 = [];
+
+	var account101 = node.randomAccount();
+	var delegates101 = [];
+
+	var transaction;
+
+	before(function () {
+		// Crediting accounts and registering delegates
```

Comment:
I would appreciate a better description - why all the steps below are needed. The scenario descriptions.